src/thor/bidirectional_astar.cc

```diff
@@ -617,11 +643,31 @@ BidirectionalAStar::GetBestPath(valhalla::Location& origin,
       return {};
     }
 
+    // Exhaust hierarchy limits simultaneously in both directions. As soon as forward/reverse
```

Comment:
@dnesbitt61 i am wondering about your comment on performance for routes that dont do hierarchy culling. where do you think that the slow down is coming from? the way this code reads to me is that we are trying to balance the searches, but wouldnt this never happen for bike and ped routes? since stop expanding would never return true? i'm just wondering where the heck the performance drop is coming, maybe its all just the updated heuristic using the actual cost. did you notice if it was more iterations?

Reply:
on my local machine I noticed that iterations increased by ~6%

Reply:
we can also take into account the fact if we use hierarchies or not

Reply:
It is more iterations on an already long search: about 6% increase on forward direction and 10% on reverse. Bicycle and pedestrian routes need a way to decrease the search space/# of iterations and I was hoping this might since the title has the word **pruning**!

Reply:
as I already mentioned this PR actually fixes a "bug". If it wasn't, we would discuss iterations decrease now